packages/gatsby/src/internal-plugins/functions/gatsby-node.ts

```diff
@@ -18,15 +19,83 @@ const isProductionEnv = process.env.gatsby_executing_command !== `develop`
 
 interface IFunctionData {
   /** The route in the browser to access the function **/
-  apiRoute: string
+  functionRoute: string
+  /** The absolute path to the original function **/
+  originalAbsoluteFilePath: string
   /** The relative path to the original function **/
-  originalFilePath: string
+  originalRelativeFilePath: string
   /** The relative path to the compiled function (always ends with .js) **/
   relativeCompiledFilePath: string
   /** The absolute path to the compiled function (doesn't transfer across machines) **/
   absoluteCompiledFilePath: string
   /** The matchPath regex created by path-to-regexp. Only created if the function is dynamic. **/
   matchPath: string
+  /** The plugin that owns this function route **/
+  pluginName: string
+}
+
+interface IGlobPattern {
+  /** The plugin that owns this namespace **/
+  pluginName: string
+  /** The root path to the functions **/
+  rootPath: string
+  /** The glob pattern **/
+  globPattern: string
+}
+
+// Create glob type w/ glob, plugin name, root path
+const createGlobArray = (siteDirectoryPath, plugins): Array<IGlobPattern> => {
```

Comment:
generate globs for the default src/api directory & one for each plugin (that's not a default plugin)